src/config.js

```diff
@@ -49,6 +48,40 @@ export async function getConfig() {
   return _config
 }
 
+/**
+ * @param {string} file
+ * @returns {Promise<LazyConfig>}
+ */
+async function loadConfigFromFile(file) {
+  if (file.endsWith('.json')) {
+    return JSON.parse(readFileSync(file, 'utf8'))
+  } else if (file.endsWith('.ts') || file.endsWith('.mts') || file.endsWith('.cts')) {
+    const { build } = await import('esbuild')
+    const result = await build({
+      absWorkingDir: workspaceRoot,
+      entryPoints: [file],
+      outfile: 'out.js',
+      target: 'esnext',
+      platform: 'node',
+      format: 'esm',
```

Comment:
Do we need to change format to `cjs` when there's config file using `cts` file extension? From what I understand esbuild should transform `module.exports` anyway.

Reply:
Yeah, I agree it likely shouldn't be part of the users' ts program, but I decided to do that since most modern tools support these extensions for the configuration file. Do you think we should keep it?

Reply:
It's fine to support the extensions yeah, but we don't need to worry about how changing how it's compiled since we are the only ones consuming the output 👍🏼